harness/src/utils.rs

```diff
@@ -128,8 +93,81 @@ pub fn kill_process_using_port(port: u16) {
         .expect("Failed to execute kill command");
 }
 
+pub async fn collect_state_matching_rafts_until_leader_emerge(
```

Comment:
I think `collect_state_matching_rafts_until_leader_emerge` is verbose and performs too many tasks.
I think it would be better to separate the functionality of returning rafts that match a specific state and waiting for a leader to emerge.